File: app/src/main/java/com/shindra/Dialog/LoadingDialog.java
package com.shindra.Dialog;

import android.app.AlertDialog;
import android.content.Context;
import android.view.LayoutInflater;

import com.shindra.R;

public class LoadingDialog extends AlertDialog {
    private AlertDialog loadingDialog;


    public LoadingDialog(Context context)
    {
        super(context);
    }


    public void startAnimation(){

    Builder builder = new Builder(getContext());
    LayoutInflater inflater = getLayoutInflater();
    builder.setView(inflater.inflate(R.layout.loading_dialog, null));
    builder.setCancelable(false);

    loadingDialog = builder.create();
    loadingDialog.show();

    }

    public void dismissDialog()
    {
        loadingDialog.dismiss();
    }
}

File: app/src/main/java/com/shindra/Dialogs/LoadingDialog.kt
package com.shindra.Dialogs

import android.content.Context
import android.view.LayoutInflater
import androidx.appcompat.app.AlertDialog
import com.shindra.R


class LoadingDialog {

    private var Dialog : AlertDialog? = null

    public fun CreateDialog (context : Context)
    {
        val DialogBuilder= AlertDialog.Builder(context)
        val dialogView= LayoutInflater.from(context).inflate(R.layout.loading_dialog,null)
        DialogBuilder.setView(dialogView)
        DialogBuilder.setCancelable(false)
        Dialog = DialogBuilder.create()
    }

    public fun ShowDialog()
    {
        Dialog?.show()
    }

    public fun CloseDialog()
    {
        Dialog?.dismiss()
    }


}